js_modules/dagit/packages/ui/src/components/MiddleTruncate.tsx

```diff
@@ -1,22 +1,46 @@
 import useResizeObserver from '@react-hook/resize-observer';
-import debounce from 'lodash/debounce';
 import * as React from 'react';
 import styled from 'styled-components/macro';
 
 import {calculateMiddleTruncation} from './calculateMiddleTruncation';
 
-interface Props {
-  text: string;
-}
-
-const getFont = (element: HTMLDivElement) => getComputedStyle(element).font;
+/**
+ * A component that performs middle truncation on a given string, based on the evaluated width
+ * of a container div.
+ *
+ * The component will render the provided string -- with no height/visibility -- into our target
+ * container to determine the maximum available width. This width value and the computed font
+ * style are then used to determine the longest middle-truncated string that can fit within the width.
+ *
+ * When the DOM element resizes, the measurement and calculation steps will occur again.
+ */
+export const MiddleTruncate: React.FC<{text: string}> = React.memo(({text}) => {
+  // Track the font style and target maximum width. `null` means no measurement has
+  // taken place yet.
+  const [truncatedText, setTruncatedText] = React.useState<string | null>(null);
+
+  // An element that renders the full text into the container, for the purpose of
+  // measuring the maximum available/necessary width for our truncated string.
+  const measure = React.useRef<HTMLDivElement>(null);
+
+  // Given the target font style and allotted width, calculate the largest possible middle-
+  // truncated string.
+  const calculateTargetStyle = React.useCallback(() => {
+    if (measure.current) {
+      setTruncatedText(calculateMiddleTruncatedText(measure.current, text));
+    }
+  }, [text]);
```

Comment:
Only playing devils advocate, but now that the component is memoized and it's only prop is `text`, I think any useEffect with just `text` in the deps can go!

It might actually be fine to `calculateMiddleTruncatedText` on every render since the entire component is memozied. (Not sure that this would be clearer though, I like being extra safe for future-proofing!)